include/dca/config/threading.hpp.in

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2009-2016 ETH Zurich
+// Copyright (C) 2007?-2016 Center for Nanophase Materials Sciences, ORNL
+// All rights reserved.
+//
+// See LICENSE.txt for terms of usage.
+// See CITATION.txt for citation guidelines if you use this code for scientific publications.
+//
+// Author: John Biddiscombe (biddisco@cscs.ch)
+//
+// This is a CMake generated header file.
+// It contains preprocessor definitions to configure the applications according to the options set
+// via CMake.
+
+#ifndef DCA_CONFIG_THREADING_DEFINES_HPP
+#define DCA_CONFIG_THREADING_DEFINES_HPP
+
+#include "dca/config/haves_defines.hpp"
+#include "dca/config/config_defines.hpp"
+
+#if defined(DCA_HAVE_HPX)
+# include "dca/parallel/hpx/hpx.hpp"
+  using Threading = dca::parallel::hpxthread;
+# ifdef DCA_HPX_MAIN
+#  include <hpx/hpx_main.hpp>
+# endif
+#else
+# include "dca/parallel/stdthread/stdthread.hpp"
+# include "dca/parallel/stdthread/thread_pool/affinity.hpp"
+  using Threading = dca::parallel::stdthread;
+#endif
+
+
```

Comment:
whether DCA_WITH_HPX=ON or OFF 
```shell
[ 99%] Building CXX object test/performance/phys/ctaux/CMakeFiles/ctaux_walker_performance_test.dir/ctaux_walker_performance_test.cpp.o
/gpfs/alpine/proj-shared/cph102/epd/DCA/test/performance/phys/ctaux/ctaux_walker_performance_test.cpp:42:45: error: conflicting declaration ‘using Threading = class dca::parallel::NoThreading’
 using Threading = dca::parallel::NoThreading;
                                             ^
In file included from /gpfs/alpine/proj-shared/cph102/epd/DCA/include/dca/phys/dca_algorithms/compute_greens_function.hpp:18,
                 from /gpfs/alpine/proj-shared/cph102/epd/DCA/include/dca/phys/dca_algorithms/compute_free_greens_function.hpp:20,
                 from /gpfs/alpine/proj-shared/cph102/epd/DCA/include/dca/phys/dca_data/dca_data.hpp:39,
                 from /gpfs/alpine/proj-shared/cph102/epd/DCA/test/performance/phys/ctaux/ctaux_walker_performance_test.cpp:25:
/gpfs/alpine/proj-shared/cph102/epd/DCA/build_hpx_summit/include/dca/config/threading.hpp:29:45: note: previous declaration as ‘using Threading = class dca::parallel::stdthread’
   using Threading = dca::parallel::stdthread;
```

```shell
[ 99%] Building CXX object test/performance/phys/ctaux/CMakeFiles/ctaux_walker_performance_test.dir/ctaux_walker_performance_test.cp\
p.o
/gpfs/alpine/proj-shared/cph102/epd/DCA/test/performance/phys/ctaux/ctaux_walker_performance_test.cpp:42:45: error: conflicting decl\
aration ‘using Threading = class dca::parallel::NoThreading’
 using Threading = dca::parallel::NoThreading;
                                             ^
In file included from /gpfs/alpine/proj-shared/cph102/epd/DCA/include/dca/phys/dca_algorithms/compute_greens_function.hpp:18,
                 from /gpfs/alpine/proj-shared/cph102/epd/DCA/include/dca/phys/dca_algorithms/compute_free_greens_function.hpp:20,
                 from /gpfs/alpine/proj-shared/cph102/epd/DCA/include/dca/phys/dca_data/dca_data.hpp:39,
                 from /gpfs/alpine/proj-shared/cph102/epd/DCA/test/performance/phys/ctaux/ctaux_walker_performance_test.cpp:25:
/gpfs/alpine/proj-shared/cph102/epd/DCA/build_hpx_summit/include/dca/config/threading.hpp:22:45: note: previous declaration as ‘usin\
g Threading = struct dca::parallel::hpxthread’
   using Threading = dca::parallel::hpxthread;
                                             ^
make[2]: *** [test/performance/phys/ctaux/CMakeFiles/ctaux_walker_performance_test.dir/ctaux_walker_performance_test.cpp.o] Error 1
make[1]: *** [test/performance/phys/ctaux/CMakeFiles/ctaux_walker_performance_test.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
I have a fix for this to PR

Reply:
I think it is due to cmake var `DCA_WITH_THREADED_SOLVER` is set to off. I am thinking to rename `Threading` to `NoThreading` for those tests are not using threading. For example, change them to`using NoThreading = dca::parallel::NoThreading;` Sorry for not testing out those tests are not using threading as I have not turned off the cmake variable.